astropy/nddata/nddata.py

```diff
@@ -5,64 +5,55 @@
                         unicode_literals)
 
 import collections
-
 import numpy as np
+from copy import deepcopy
 
 from .nddata_base import NDDataBase
+from .nduncertainty import NDUncertainty
 from ..units import Unit, Quantity
 from .. import log
 from ..utils.compat.odict import OrderedDict
-
-from ..config import ConfigAlias
+from ..extern import six
 
 __all__ = ['NDData']
 
-
 __doctest_skip__ = ['NDData']
 
-
-WARN_UNSUPPORTED_CORRELATED = ConfigAlias(
-    '0.4', 'WARN_UNSUPPORTED_CORRELATED', 'warn_unsupported_correlated',
-    'astropy.nddata.nddata', 'astropy.nddata')
-
-
 class NDData(NDDataBase):
     """
     A basic class for array-based data.
 
-    The key distinction from raw numpy arrays is the presence of
+    The key distinction from raw `numpy.ndarray` is the presence of
     additional metadata such as uncertainties, a mask, units,
     and/or a coordinate system.
 
     Parameters
     -----------
     data : `~numpy.ndarray`, `~numpy.ndarray`-like, or `NDData`
-        The actual data contained in this `NDData` object. If possible, data
-        will not be copied`data`, so you should make copy the ``data`` before
-        passing it in if that's the desired behavior.
+        The actual `data` contained in this `NDData` object.
 
     uncertainty : any type, optional
-        Uncertainty on the data. The uncertainty *must* have a string attribute
-        named ``uncertainty_type``, but there is otherwise no restriction.
+        Uncertainty on the data. The `uncertainty` *should* have a string
+        attribute named ``uncertainty_type``, but there is otherwise no
+        restriction.
 
-    mask : `~numpy.ndarray`-like, optional
-        Mask for the data. The values must be ``False`` where
-        the data is *valid* and ``True`` when it is not (like Numpy
-        masked arrays). If ``data`` is a numpy masked array, providing
-        ``mask`` here will causes the mask from the masked array to be
-        ignored.
+    mask : any type, optional
+        Mask for the data.
```

Comment:
Definitely want the deleted comments about behaving like a numpy mask restored here....